### PR TITLE
Force editor progress dialog size update.

### LINKED ISSUE #92828: Progress popup window height isn't updating properly
### Tested versions

v4.3.beta1.official [a4f2ea91a]
But it's been like this in 4.2 and 4.1 as well

### System information

Godot v4.3.beta1 - macOS 14.4.1 - Vulkan (Forward+) - integrated Apple M1 Max - Apple M1 Max (10 Threads)

### Issue description

A bit of a nitpick I guess but the import progress popup is getting squashed when a second progressbar is added to it:

image<br>

This is what it looks like in a video:

https://github.com/godotengine/godot/assets/60579014/46f4f333-087b-4f22-a4e4-7afe1e59204f

### Steps to reproduce

I suppose import some gltf file and some texture files at the same time to see this

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/92828

### CODE REVIEW

**Comment by passivestar** [comment]
Tested, it works

File: editor/progress_dialog.cpp [hunk 1]
```diff
@@ -147,6 +157,12 @@ void ProgressDialog::_popup() {
 		}
 		// No host window found, use main window.
 		EditorInterface::get_singleton()->popup_dialog_centered(this, ms);
+	} else {
+		Rect2i adjust = _popup_adjust_rect();
+		if (adjust != Rect2i()) {
+			set_position(adjust.position);
+			set_size(adjust.size);
+		}
```
Comment (adamscott) [inline editor/progress_dialog.cpp:165]: I would invert the condition for better readability (to prevent a double negative).

**Comment by adamscott** [comment]
Is the `Main::iteration()` (2 loops) call prevents the loading from happening? I'm curious.

**Comment by bruvzg** [comment]
> Is the Main::iteration() (2 loops) call prevents the loading from happening? I'm curious.

Two loops do to fix it, some value might do (10 iterations seems to), but it's better to just update rect directly.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by SoloWingPixy2097** [comment]
Maybe it's just me, but isn't this fix a bit janky, to call the entire Main::Iteration() again, from within a Main::iteration()? (Essentially a recursive call)

In my project, this ends up breaking ImGui integration, because imgui relies on a single start and end of frame call (which we have bound to the Main::Iteration()), so doing another Main::Iteration() results in doubling up the start and end calls, triggering a SIGABRT (From an Imgui Assert)

I understand my problem is pretty specific, but anyone relying on Main::Iteration() being exactly that, 1 iteration, will get caught off guard by this

I can work around this by undoing this commit locally, because the little graphical squish is much less of an issue than having literally everything re-invoked, but this seems counter productive for future Godot Version Integrations
Alternatively I can keep track of the current scope and guard against recursive invokes, which is what I've opted to do for the time being

It seems a little odd to me that fixing a GUI squish requires an entire engine tick to fix, granted this does appear to be the workflow in a couple other places too :/
Unfortunately I'm not familiar enough with Godot's GUI source code to provide a better alternative, but hoping to bring up discussion about potential other solutions?

**Pull request merged**